packages/common-all/src/dnode.ts

```diff
@@ -1762,8 +1782,10 @@ export class SchemaUtils {
     const match = _.find(schemas, (sc) => {
       const pattern = SchemaUtils.getPatternRecursive(sc, schemaModule.schemas);
       if (sc?.data?.namespace && matchNamespace) {
+        // if we are matching a namespace node,
+        // match on the level of the node itself and its immediate children.
         namespace = true;
-        return minimatch(notePathClean, _.trimEnd(pattern, "/*"));
```

Comment:
The issue here was that `_.trimEnd` was removing the entire glob pattern when what we really wanted was to trim only one level of it.

Reply:
its not immediately obvious what `slice(0,-2)` means - can you add some comments here?

Reply:
improved the comments to make clear what's happening here.